src/persist-client/src/cli/admin.rs

```diff
@@ -125,6 +130,50 @@ pub async fn run(command: AdminArgs) -> Result<(), anyhow::Error> {
             .await?;
             info_log_non_zero_metrics(&metrics_registry.gather());
         }
+        Command::RestoreBlob(args) => {
+            let StoreArgs {
+                consensus_uri,
+                blob_uri,
+            } = args;
+            let commit = command.commit;
+            let cfg = PersistConfig::new(&BUILD_INFO, SYSTEM_TIME.clone());
+            let metrics_registry = MetricsRegistry::new();
+            let metrics = Arc::new(Metrics::new(&cfg, &metrics_registry));
+            let consensus =
+                make_consensus(&cfg, &consensus_uri, commit, Arc::clone(&metrics)).await?;
+            let blob = make_blob(&cfg, &blob_uri, commit, Arc::clone(&metrics)).await?;
+            let versions = StateVersions::new(
+                cfg.clone(),
+                Arc::clone(&consensus),
+                Arc::clone(&blob),
+                metrics,
+            );
+
+            let mut shards = consensus.list_keys();
+            let mut not_restored = vec![];
+            while let Some(shard) = shards.next().await {
+                let shard_id = shard?;
+                let shard_id = ShardId::from_str(&shard_id).expect("invalid shard id");
+
+                let shard_not_restored = crate::internal::restore::restore_blob(
+                    &versions,
+                    blob.as_ref(),
+                    &cfg.build_version,
+                    shard_id,
+                )
+                .await?;
+                info!(
```

Comment:
nit: let's put a log before, too, so we know what shard it's currently working on. also may as well stick the time it took in the finished log message